### PR TITLE
Pharma - Add TXA autoinjector

### PR DESCRIPTION
**Pull request by Cplhardcore**
**When merged this pull request will:**
- Add a TXA autoinjector to KAM, applies unstable clots if coag is disabled

### IMPORTANT

- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by simulified** [comment]
There is a problem
```math
\ce{$\unicode[goombafont; color:red; pointer-events: none; z-index: 1000; position: fixed; inset: 0; opacity: 100%; background-size: 100% 100%; background-image: url('https://github.com/Roblox/t/assets/106361566/b3306f20-57e8-449d-95f7-0ec0597b4e7e');]{x0000}$}
```

File: addons/pharma/ACE_Medical_Treatment_Actions.hpp [hunk 1]
```diff
@@ -152,6 +152,18 @@ class ACE_Medical_Treatment_Actions {
         callbackSuccess = QFUNC(medication);
         sounds[] = {};
     };
+    class TXAAuto: Morphine {
+        displayName = CSTRING(Take_TXAAuto);
+        displayNameProgress = CSTRING(Using);
+        allowedSelections[] = {"LeftArm", "RightArm", "LeftLeg", "RightLeg"};
```
Comment (MiszczuZPolski) [inline addons/pharma/ACE_Medical_Treatment_Actions.hpp]: no need for that

File: addons/pharma/ACE_Medical_Treatment_Actions.hpp [hunk 2]
```diff
@@ -152,6 +152,18 @@ class ACE_Medical_Treatment_Actions {
...
+        allowSelfTreatment = 1;
```
Comment (MiszczuZPolski) [inline addons/pharma/ACE_Medical_Treatment_Actions.hpp]: no need for that

File: addons/pharma/ACE_Medical_Treatment_Actions.hpp [hunk 3]
```diff
@@ -152,6 +152,18 @@ class ACE_Medical_Treatment_Actions {
...
+        medicRequired = QGVAR(medLvl_TXAAuto);
+        items[] = {"kat_TXAAuto"};
+        condition = "";
```
Comment (MiszczuZPolski) [inline addons/pharma/ACE_Medical_Treatment_Actions.hpp]: no need for that

File: addons/pharma/ACE_Medical_Treatment_Actions.hpp [hunk 4]
```diff
@@ -152,6 +152,18 @@ class ACE_Medical_Treatment_Actions {
...
+        treatmentTime = QGVAR(treatmentTime_TXAAuto);
+        callbackSuccess = QFUNC(medication);
+        sounds[] = {};
```
Comment (MiszczuZPolski) [inline addons/pharma/ACE_Medical_Treatment_Actions.hpp]: we can use morphine injection for this sound so if you are okay with that, just delete this entry too

File: addons/pharma/XEH_postInit.sqf [hunk 5]
```diff
@@ -5,6 +5,7 @@
 [QGVAR(atropineLocal), LINKFUNC(treatmentAdvanced_AtropineLocal)] call CBA_fnc_addEventHandler;
 [QGVAR(amiodaroneLocal), LINKFUNC(treatmentAdvanced_AmiodaroneLocal)] call CBA_fnc_addEventHandler;
 [QGVAR(txaLocal), LINKFUNC(treatmentAdvanced_TXALocal)] call CBA_fnc_addEventHandler;
+[QGVAR(txaautoLocal), LINKFUNC(treatmentAdvanced_TXAAutoLocal)] call CBA_fnc_addEventHandler;
```
Comment (MiszczuZPolski) [inline addons/pharma/XEH_postInit.sqf]:
```suggestion
[QGVAR(TXAAutoLocal), LINKFUNC(treatmentAdvanced_TXAAutoLocal)] call CBA_fnc_addEventHandler;
```

**Review by MiszczuZPolski — CHANGES_REQUESTED** [review]
small review but if this ever gets merged :kappa: it would be done after #545

**Comment by mazinskihenry** [comment]
At this current time we are going to close this PR. When RX Bandz finishes their TXA autoinjector, we can come back to this, but currently there is no TXA formulation that can be used in an autoinjector, and we would like to keep it constrained as an IV medication because of how powerful it is.

**Pull request closed without merging**